docs/content/sips/012-signing-spin-releases.md

```diff
@@ -0,0 +1,210 @@
+a private key
+
+The keyless mode of Cosign can eliminate the key management issue through
+ephemeral keys, which is why this method is preferred to alternatives.
+
+### Security considerations
+
+[Sigstore's trust model](https://docs.sigstore.dev/security/) relies on a few
+key trusted components: the OIDC provider to prove identity, the trust root
+based on [TUF](https://theupdateframework.io/), [Rekor](https://docs.sigstore.dev/rekor/overview/),
+and [Fulcio](https://docs.sigstore.dev/fulcio/overview/).
+
+If any of the components above are compromised, the security guarantees of _any_
+signature is no longer valid. However, because of the transparency logs,
+such compromises can be detected.
+
+Unauthorized access to the GitHub actions used to generate the releases could
+allow an attacker to tamper with the release — however, that attack surface is
+present regardless of what signature mechanism we choose — which suggests increased
+attention to the workflows used to build and generate release artifacts.
+
+### Appendix: minimal example in GitHub Actions
+
+[The following repository](https://github.com/radu-matei/keyless-cosign-demo) can
+be used as a minimal example for this workflow:
+
+```yml
+jobs:
+  sign:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write 
+
+    name: Sign artifact and publish signature and certificate
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@main
+        with:
+          cosign-release: v2.0.0
+
+      - name: Create an artifact
+        run: |
+          mkdir out
+          echo 'hello world' > out/artifact
+          
+      - name: Sign the artifact with GitHub OIDC token
+        run: cosign sign-blob --output-certificate out/crt.pem --output-signature out/artifact.sig out/artifact --yes
+
+      - name: Upload assets as GitHub artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: artifact
+          path: out/*
+      ...(continue with adding artifacts to the release)
+```
+
+### Verifying signatures
+
+After the action is executed, the new artifact is released and its signature
+published using Sigstore. To validate the signature, a user will then need to
+download the release package (which now contains the artifact itself, the
+certificate used to sign, and the signature itself), then validate the signature
+using `cosign`:
+
+```bash
+$ tree .
+├── artifact
+├── artifact.sig
+└── crt.pem
+
+$ cosign verify-blob \
+    --signature artifact.sig --certificate crt.pem \
+    # the identity for the certificate is tied to the official repository of the Spin project
+    --certificate-identity https://github.com/fermyon/spin/.github/workflows/release-sign.yml@refs/heads/main \
+    --certificate-oidc-issuer https://token.actions.githubusercontent.com \
+    artifact
+
+Verified OK
```

Comment:
would be nice to have a spin self update command or plugin that handles this verification part for the user if possible in the future.